src/main/java/prefs/UserConfig.java

```diff
@@ -0,0 +1,7 @@
+package prefs;
+
+/**
+ * Abstraction for user-defined settings.
+ */
+public class UserConfig {
+}
```

Comment:
But preferences will be heavily affected. All its creation methods take in both the session and user config. Since we are already heading in that direction, we might as well create it now, during the refactoring. It's not really "future use". I feel that it's part of the new preferences structure to have both session and user config files present and separate from each other. The goal of this PR was to enable the config/preferences code to handle more than one global(session) config anyway.

I realise that there is no code in the user config now. However it is not trivial to add it later, as the preferences has to change a lot.